src/main/java/edu/harvard/iq/dataverse/UserNotification.java

```diff
@@ -30,7 +37,48 @@
         ASSIGNROLE, REVOKEROLE, CREATEDV, CREATEDS, CREATEACC, SUBMITTEDDS, RETURNEDDS, 
         PUBLISHEDDS, REQUESTFILEACCESS, GRANTFILEACCESS, REJECTFILEACCESS, FILESYSTEMIMPORT, 
         CHECKSUMIMPORT, CHECKSUMFAIL, CONFIRMEMAIL, APIGENERATED, INGESTCOMPLETED, INGESTCOMPLETEDWITHERRORS, 
-        PUBLISHFAILED_PIDREG, WORKFLOW_SUCCESS, WORKFLOW_FAILURE, STATUSUPDATED, DATASETCREATED
+        PUBLISHFAILED_PIDREG, WORKFLOW_SUCCESS, WORKFLOW_FAILURE, STATUSUPDATED, DATASETCREATED;
+        
+        public String getDescription() {
+            return BundleUtil.getStringFromBundle("notification.typeDescription." + this.name());
+        }
+        
+        public long flagValue() {
+            return 1 << this.ordinal();
+        }
+        
+        public static List<Type> fromFlag(Long flag) {
```

Comment:
While I generally like the idea of using a custom made serialization of using a `long` for sth similar to a bitfield, this is more C than Java.

How about using `Set<Type>` directly? (Exposing an `EnumSet`) You could use `@PrePersist` and `@PostLoad` if you still want to serialize into a single database field, but make it encapsuled and transparent to users of the model class.

Reply:
I have implemented your suggestion. I am not entirely sure if I did it exactly as you suggested, I am little bit concerned with the setters in the `AuthenticatedUser`. I still have a setter for the long value since it is used by a query. Since I needed to keep both types of setters (for a `Set` and a `long`), I need to keep the values in sync. Would you mind to look at this again? Thank you.

Reply:
I would, but could you please push your commits? Thx :smile:

Reply:
Sorry :blush:

Reply:
I had some problems with git, lost my changes.... Finally it is working again. The configuration is much better now with setting wrapper.